server/athenian/api/internal/features/entries.py

```diff
@@ -1579,6 +1558,90 @@ def _mapper(self) -> JIRAEntitiesMapper:
         return self._entities_mapper
 
 
+@dataclasses.dataclass
+class _FactsJIRAFiltersHelper:
+    """Collect the needed information to correctly mine and group facts by JIRAFilter-s."""
+
+    filter_: JIRAFilter
+    """The single filter to use to mine facts."""
+    entities_to_fetch: JIRAEntityToFetch
+    """The entities to fetch in order to correctly apply grouping on mined facts"""
+    grouper: Callable[[pd.DataFrame], list[np.ndarray]]
+    """The grouping function to group by jira filters."""
+
+
+class _PRsJIRAFiltersHelper(_FactsJIRAFiltersHelper):
+    """Collect the needed information to correctly fetch and group prs by JIRAFilter-s."""
+
+    @classmethod
+    async def build(
+        cls,
+        jiras: Sequence[JIRAFilter],
+        metrics: Sequence[str],
+        account: int,
+        mdb: Database,
+        cache: aiomcache.Client | None,
+    ) -> _PRsJIRAFiltersHelper:
+        filter_ = JIRAFilter.combine(*jiras)
+        if len(jiras) == 1:
```

Comment:
This way we have the previous performance, before `jira` => `jiras`, when there's a single JIRAFilter